packages/xstate-store/src/index.ts

```diff
@@ -0,0 +1,220 @@
+>(
+  initialContext: TContext,
+  transitions: {
+    [K in keyof TEventPayloadMap & string]:
+      | StoreAssigner<NoInfer<TContext>, { type: K } & TEventPayloadMap[K]>
+      | StorePropertyAssigner<
+          NoInfer<TContext>,
+          { type: K } & TEventPayloadMap[K]
+        >;
+  },
+  updater?: (
+    context: NoInfer<TContext>,
+    recipe: (context: NoInfer<TContext>) => NoInfer<TContext>
+  ) => NoInfer<TContext>
+): Store<TContext, ExtractEventsFromPayloadMap<TEventPayloadMap>> {
+  type StoreEvent = ExtractEventsFromPayloadMap<TEventPayloadMap>;
+  const setter = updater ?? defaultSetter;
+  let observers: Set<Observer<StoreSnapshot<TContext>>> | undefined;
+  const initialSnapshot: StoreSnapshot<TContext> = {
+    context: initialContext,
+    status: 'active'
+  };
+  let currentSnapshot = initialSnapshot;
+
+  function receive(event: StoreEvent) {
+    let currentContext = currentSnapshot.context;
+    const assigner = transitions?.[event.type as StoreEvent['type']];
+
+    if (!assigner) {
+      return;
+    }
+
+    if (typeof assigner === 'function') {
+      currentContext = updater
+        ? updater(
+            currentContext,
+            (draftContext) =>
+              (assigner as StoreCompleteAssigner<TContext, StoreEvent>)?.(
+                draftContext,
+                event
+              )
+          )
+        : setter(currentContext, (draftContext) =>
+            Object.assign(
+              {},
+              currentContext,
+              assigner?.(
+                draftContext,
+                event as any // TODO: help me
+              )
+            )
+          );
+    } else {
+      const partialUpdate: Record<string, unknown> = {};
+      for (const key of Object.keys(assigner)) {
+        const propAssignment = assigner[key];
+        partialUpdate[key] =
+          typeof propAssignment === 'function'
+            ? (
+                propAssignment as StorePartialAssigner<
+                  TContext,
+                  StoreEvent,
+                  typeof key
+                >
+              )(currentContext, event)
+            : propAssignment;
+      }
+      currentContext = Object.assign({}, currentContext, partialUpdate);
+    }
+
+    currentSnapshot = { ...currentSnapshot, context: currentContext };
+
+    observers?.forEach((o) => o.next?.(currentSnapshot));
+  }
+
+  const store: Store<TContext, StoreEvent> = {
+    send(event) {
+      receive(event as unknown as StoreEvent);
+    },
+    getSnapshot() {
+      return currentSnapshot;
+    },
+    getInitialSnapshot() {
+      return initialSnapshot;
+    },
+    subscribe(observerOrFn) {
+      const observer = toObserver(observerOrFn);
+      observers ??= new Set();
+      observers.add(observer);
+
+      return {
+        unsubscribe() {
+          return observers?.delete(observer);
+        }
+      };
+    },
+    [symbolObservable](): InteropSubscribable<StoreSnapshot<TContext>> {
+      return this;
+    }
```

Comment:
if you decouple this pkg from `xstate` then to make this work you should also include smth like this at the bottom of this file or somewhere:
https://github.com/statelyai/xstate/blob/f86a59ced85febed3ee67ea2bbe28fd34c465c6d/packages/core/src/index.ts#L42-L46